sam_view.c

```diff
@@ -345,13 +351,29 @@ int main_samview(int argc, char *argv[])
         //case 'x': out_format = "x"; break;
         //case 'X': out_format = "X"; break;
                  */
-        case '?': is_long_help = 1; break;
+        case '?':
+            if (optopt == '?') {  // '-?' appeared on command line
```

Comment:
This all feels excessively complicated and obfuscated for what is, when all said and done, a rather pointless feature that is probably better done by changing `-?` to `--help`.

If we really want to keep it as -? then maybe something far simpler (as suggested by the same stackoverflow answer) works:

```
-        case '?': is_long_help = 1; break;
+        case '?':
+            if (optind>0 && !strcmp(argv[optind-1],"-?"))
+                is_long_help = 1;
+            else
+                return usage(stderr, EXIT_FAILURE, 0);
+            break;
```

Yes it doesn't cope with bizarreness like `samtools view -h?`, but really I don't see this being something we should go out of our way to support.  My preference is still just drop the use of `-?` completely as it's just asking for confusion.